gcp_deepvariant_runner.py

```diff
@@ -1040,10 +1102,57 @@ def run(argv=None):
             'jobs. By default, the pipeline runs all 3 jobs (make_examples, '
             'call_variants, postprocess_variants) in sequence. '
             'This option may be used to run parts of the pipeline.'))
+  parser.add_argument(
+      '--stop_collecting_anonymous_usage_metrics',
+      default=False,
+      action='store_true',
+      help=('Use preemptible VMs for the pipeline.'))
 
   pipeline_args = parser.parse_args(argv)
   _validate_and_complete_args(pipeline_args)
 
+  def get_image_version(docker_image):
+    return docker_image.split(':')[-1]
+
+  def get_model_version(model):
+    return model.split('/')[-1]
+
+  if not pipeline_args.stop_collecting_anonymous_usage_metrics:
+    metrics.add(
+        _get_project_number(pipeline_args.project),
```

Comment:
I am not sure whether it still makes sense to collect the metrics when getting the project number failed (-1).

Reply:
Do you know in which case it may fail to get the project_id?

Reply:
I can imagine there *might* be some GCP IAM settings that prevents a service account from reading the project's metadata, I am guessing in those cases it will also raise an error.
It is also possible we can successfully fetch the project_id in all runs and never return -1. But in any case in _get_project_number()  we have to run the `gcloud projects describe` command in a try-catch block.

Reply:
According to `gcloud projects describe --help`:

> This command can fail for the following reasons:
>  * The project specified does not exist.
>  * The active account does not have permission to access the given project

The first case does not occur to us (otherwise the `gcloud alpha genomics pipelines run` would't go through). So we will not log the right project_id when serviceaccount does not have enough permission.

Reply:
In general we aren't allowed to log project ID (it's PII).  I think we got approval to log a SHA1 hash of the project ID.

Note that the permissions required to get the project ID are non-trivial and so it might be likely that the SA doesn't have them.

Reply:
Thank you Saman for the detailed explanation! I think the second case should not occur to us neither, since the user is running the pipeline in that project, which they must have access to. So I agree that it probably will never return -1 :).

Reply:
Sorry I misread this as ID not number for some reason.  Project number is fine :)